cmd/bundle/generate/utils.go

```diff
@@ -0,0 +1,75 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/notebook"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"gopkg.in/yaml.v3"
+)
+
+func saveConfigToFile(ctx context.Context, data any, filename string) error {
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	err = encode(data, file)
+	if err != nil {
+		return err
+	}
+
+	cmdio.LogString(ctx, fmt.Sprintf("Job configuration successfully saved to %s", filename))
+	return nil
+}
+
+func encode(data any, w io.Writer) error {
+	enc := yaml.NewEncoder(w)
+	enc.SetIndent(2)
+	return enc.Encode(data)
+}
+
+func downloadNotebookAndReplaceTaskPath(ctx context.Context, task *jobs.Task, w *databricks.WorkspaceClient, outputDir string) error {
+	if task.NotebookTask == nil {
+		return nil
+	}
+
+	info, err := w.Workspace.GetStatusByPath(ctx, task.NotebookTask.NotebookPath)
+	if err != nil {
+		return err
+	}
+
+	ext := notebook.GetExtensionByLanguage(info)
+
+	reader, err := w.Workspace.Download(ctx, task.NotebookTask.NotebookPath)
+	if err != nil {
+		return err
+	}
+
+	filename := path.Base(task.NotebookTask.NotebookPath) + ext
+	targetPath := filepath.Join(outputDir, filename)
+	f, err := os.Create(targetPath)
+	if err != nil {
```

Comment:
What should we do if the file already exists? E.g. when a user runs this command multiple times.

Reply:
@pietern right now the behaviour is to always overwrite it. We have 2 options here:
1. Make this more clear by showing the warning to the customer stating that files will be overiden
2. Do not override by default, fail if there are files and add "--force" flag to force overriding.

I'm leaning towards Option 2